scripts/rv_expression_association/get_vep_variants.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import logging
+import hail as hl
+import numpy as np
+import pandas as pd
+from hail.methods import export_plink
+from cpg_utils.hail_batch import dataset_path, init_batch, output_path
+
+# object containing variants within a 50K window on either side of the VPREB3 gene
+MT = dataset_path('v0/VPREB3_50K_window.mt')
+VEP_HT = dataset_path('v0/VPREB3_50K_window_vep.ht')
+
+# use logging to print statements, display at info level
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+
+
+def main():
+    # read and densify object
+    init_batch()
+    mt = hl.read_matrix_table(MT)
+    mt = hl.experimental.densify(mt)
+
+    logging.info('Number of variants in window: {}'.format(mt.count()[0]))
+
+    # filter out low quality variants
+    mt = mt.filter_rows(hl.len(hl.or_else(mt.filters, hl.empty_set(hl.tstr))) == 0)
+
+    # consider biallelic variants only (no multi-allelic, no ref-only)
+    mt = mt.filter_rows(hl.len(mt.alleles) == 2)
+    mt = mt.filter_rows(hl.is_snp(mt.alleles[0], mt.alleles[1]))
+
+    # annotate using VEP
+    vep_ht = hl.read_table(VEP_HT)
+    mt = mt.annotate_rows(vep=vep_ht[mt.row_key].vep)
+
+    logging.info('Number of QC-passing, biallelic SNPs: {}'.format(mt.count()[0]))
+
+    # filter rare variants only (MAF < 5%)
+    mt = hl.variant_qc(mt)
+    rv_mt = mt.filter_rows(mt.variant_qc.AF[1] < 0.05)
+    rv_mt = rv_mt.filter_rows(rv_mt.variant_qc.AF[1] > 0)
+
+    logging.info('Number of rare variants (freq<5%): {}'.format(rv_mt.count()[0]))
+
+    # filter variants found to have regulatory effects
+    filtered_mt = rv_mt.filter_rows(
+        hl.len(rv_mt.vep.regulatory_feature_consequences['biotype']) > 0
+    )
+    logging.info(
+        'Number of rare variants (freq<5%) with ergulatory conequences: {}'.format(
+            filtered_mt.count()[0]
+        )
+    )
+
+    filtered_rrv_mt = filtered_mt.filter_rows(filtered_mt.variant_qc.AF[1] < 0.01)
+    logging.info(
+        'Number of rarer variants (freq<1%): {}'.format(filtered_rrv_mt.count()[0])
+    )
+
+    filtered_0maf_mt = filtered_mt.filter_rows(filtered_mt.variant_qc.AF[1] == 0)
+    logging.info(
+        'Check that there are {} variants with freq==0'.format(
+            filtered_0maf_mt.count()[0]
+        )
+    )
+
+    # print out stats
+    # types of regulatory variants
+    biotypes = pd.Series(
+        filtered_mt.vep.regulatory_feature_consequences['biotype'].collect()
+    )
+    print(
+        biotypes.value_counts()
+    )  # not sure how to turn this into logging? Multiline table
```

Comment:
Do you need to log this at all?

Reply:
same as above, some biotypes are more likely regulatory than others, so useful to explore this in a few cases to inform my strategy - but again I could just save this information in a table instead of printing these logs, if they are very expensive

Reply:
(See my comment above -- it's not the logs itself (although a table might be more handy for post-processing in a notebook), but the repeated evaluation.)